src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1660,8 +1669,10 @@ private static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSta
 
         private static BoundStatement BindConstructorInitializerIfAny(MethodSymbol method, TypeCompilationState compilationState, DiagnosticBag diagnostics)
         {
+            Debug.Assert(!method.IsExtern);
+
             // delegates have constructors but not constructor initializers
-            if (method.MethodKind == MethodKind.Constructor && !method.ContainingType.IsDelegateType() && !method.IsExtern)
+            if (method.MethodKind == MethodKind.Constructor && !method.ContainingType.IsDelegateType())
```

Comment:
This should be an assert, because it should now be impossible for the constructor of a delegate type to get here.

Reply:
I agree that it's impossible for a constructor of a delegate type to get here, but that's not what we're checking. The condition is satisfied when we have a constructor of a non-delegate type.

Reply:
So it's always true?

Reply:
Or, at least, the second part is always true?

Reply:
Or, at least, the second part is always true?

Reply:
Second part moved to assert.